compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -408,6 +408,86 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             err.downgrade_to_delayed_bug();
         }
 
+        if let (ty::Adt(adt_def, _), SelfSource::QPath(_)) = (rcvr_ty.kind(), source) {
```

Comment:
please pull this out into a method. Also, can you use a let-chain here instead of destructuring a tuple?

Reply:
I tend to use tuple destructuring in favor of let-chains when I want to make sure rustfmt works, particularly in long blocks. I can change it, of course. I personally like let-chains more.